packages/samples/headless-ssr-commerce/src/lib/navigatorContextProvider.ts

```diff
@@ -0,0 +1,32 @@
+import {NavigatorContext} from '@coveo/headless/ssr-commerce';
+import type {ReadonlyHeaders} from 'next/dist/server/web/spec-extension/adapters/headers';
+
+export class NextJsNavigatorContext implements NavigatorContext {
+  constructor(private headers: ReadonlyHeaders) {}
+
+  get referrer() {
+    return this.headers.get('referer') || this.headers.get('referrer'); // Some browsers use 'referer'
+  }
+
+  get userAgent() {
+    return this.headers.get('user-agent');
+  }
+
+  get location() {
+    return 'TODO:';
+  }
+
+  get clientId() {
```

Comment:
If the client ID gets generated here, it would have to be set (through a Relay util function?) and then sent back in the custom header on subsequent requests.

Is such a util function already available in Relay? If not, do we have a Jira for this?

I think we should add a comment here and/or a TODO.